sympy/physics/quantum/grover.py

```diff
@@ -305,7 +331,8 @@ def apply_grover(oracle, nqubits, iterations=None):
         >>> from sympy.physics.quantum.qapply import qapply
         >>> from sympy.physics.quantum.qubit import IntQubit
         >>> from sympy.physics.quantum.grover import apply_grover
-        >>> f = lambda qubits: qubits == IntQubit(2)
+        >>> from sympy.physics.quantum.grover import OracleGateFunction
+        >>> f = OracleGateFunction(lambda qubits: qubits == IntQubit(2))
         >>> qapply(apply_grover(f, 2))
```

Comment:
Maybe `apply_grover` or `OracleGate` could do this conversion automatically.

Reply:
I thought about that, but I couldn't find the right `__new__` function where this would make sense but I am open to suggestions. I also considered creating a generic `PythonFunctionWrapper` and add it to `_sympy_converter`, but felt it was a bit invasive/encouraging to write sloppy code

Reply:
It's probably not worth overthinking this. If the quantum module becomes maintained then hopefully someone can come up with sensible ideas.

Reply:
I think I've found the right place to do this automatically.